scripts/watchBackedUpNode.js

```diff
@@ -0,0 +1,132 @@
+var readFile = promisify(fs.readFile);
+var ethers = require("ethers");
+
+const PORT1 = "8543";
+const PORT2 = "8546";
+const RETRY_TIMEOUT_SECONDS = 2;
+const SCAN_INTERVAL_SECONDS = 5;
+const ARTIFACTS_PATH = "../posdao-contracts/build/contracts/";
+const PASSWORD_PATH = "/../config/password"
+const SIGNER_ADDRESS1 = "0x522df396ae70a058bd69778408630fdb023389b2";
+const SIGNER_ADDRESS2 = "0x522df396ae70a058bd69778408630fdb023389b2";
+const DUMMY_SIGNER_ADDRESS = "0x69103d3decf4a462e7f95b36684b04f86a950028";
+
+var Web3 = require("web3");
+var web3_1 = new Web3(new Web3.providers.HttpProvider(`http://localhost:${PORT1}`));
+var web3_2 = new Web3(new Web3.providers.HttpProvider(`http://localhost:${PORT2}`));
+var provider = new ethers.providers.JsonRpcProvider(`http://localhost:${PORT2}`);
+// `true` if the primary is required to sign and `false` if the secondary does.
+var primaryHasToSign = true;
+var validatorSetContract = new web3_2.eth.Contract(
+  require(`${ARTIFACTS_PATH}ValidatorSetAuRa.json`).abi,
+  '0x1000000000000000000000000000000000000001'
+);
+
+async function scanBlocks(depth) {
+    assert(typeof depth === "number");
+    var lastBlock = await web3_2.eth.getBlock("latest");
+    var lastBlockNum = lastBlock.number;
+    assert(typeof lastBlockNum === "number");
+    if (lastBlockNum < depth) {
+        return true;
+    }
+    var startBlockNum = lastBlockNum - depth;
+
+    console.log(`Scanning blocks from ${startBlockNum} to ${lastBlockNum}`);
+
+    for (var i = startBlockNum;  i <= lastBlockNum; i++) {
+        let block = await web3_2.eth.getBlock(i);
+        if (block.author === SIGNER_ADDRESS) {
+            return true;
+        }
+    }
+    return false;
+}
+
+// Starts signing at the secondary node by setting the secondary signer address.
+async function startSecondarySigning() {
+    console.log(`Reserve node at port ${PORT2} starts signing`);
+
+    let password = await readFile(`__dirname${PASSWORD_PATH}`, "UTF-8");
+    assert(typeof password === "string");
+    await provider.send(
+        "parity_setEngineSigner",
+        [ SIGNER_ADDRESS2, password.trim() ]
+    );
+}
+
+// Stops signing at the secondary node by setting the dummy signer address.
+async function stopSecondarySigning() {
+    console.log(`Reserve node at port ${PORT2} stops signing`);
+
+    let password = await readFile(`__dirname${PASSWORD_PATH}`, "UTF-8");
+    assert(typeof password === "string");
+    await provider.send(
+        "parity_setEngineSigner",
+        [ DUMMY_SIGNER_ADDRESS, password.trim() ]
+    );
+}
+
+async function startScan() {
+    var secondaryListening = false;
+    try {
+        secondaryListening = await web3_2.eth.net.isListening();
+    } catch(e) {
+        console.log("Disconnected from secondary");
+    }
+    assert(typeof secondaryListening === "boolean");
+    if (secondaryListening) {
+        let validators = await validatorSetContract.methods.getValidators().call();
```

Comment:
It's probably best to add a line
```js
validators = validators.map(v => v.toLowerCase());
```
after this, and also do case-insensitive comparison below
```js
if (validators.indexOf(SIGNER_ADDRESS1.toLowerCase()) != -1)
```
otherwise it may not work

Reply:
Done.